packages/bls-kms-local/src/agent/BlsKeyManagementSystem.ts

```diff
@@ -0,0 +1,97 @@
+import {KeyManagementSystem} from "@veramo/kms-local"
+
+const debug = Debug('veramo:kms:bls:local')
+
+export class BlsKeyManagementSystem extends KeyManagementSystem {
+
+  private readonly privateKeyStore: AbstractPrivateKeyStore;
+
+  constructor(keyStore: AbstractPrivateKeyStore) {
+    super(keyStore);
+    this.privateKeyStore = keyStore;
+  }
+
+  async importKey(args: Exclude<MinimalImportableKey, 'kms'>): Promise<ManagedKeyInfo> {
+    if (args.type === <TKeyType>'Bls12381G2') {
+      if (!args.type || !args.privateKeyHex || !args.publicKeyHex) {
+        throw new Error('invalid_argument: type, publicKeyHex and privateKeyHex are required to import a key')
+      }
+      const managedKey = this.asBlsManagedKeyInfo({ alias: args.kid, privateKeyHex: args.privateKeyHex, publicKeyHex: args.publicKeyHex, type: args.type });
+      await this.privateKeyStore.import({alias: managedKey.kid, ...args})
+      debug('imported key', managedKey.type, managedKey.publicKeyHex)
+      return managedKey
+    }
+    return super.importKey(args) as Promise<ManagedKeyInfo>
+  }
+
+  async createKey({ type }: { type: TKeyType }): Promise<ManagedKeyInfo> {
+    let key: ManagedKeyInfo
+    switch (type) {
+      case <TKeyType>'Bls12381G2': {
+        const keyPairBls12381G2 = await generateBls12381G2KeyPair()
+        key = await this.importKey({
+          kms: 'local',
+          type,
+          privateKeyHex: Buffer.from(keyPairBls12381G2.secretKey).toString("hex"),
+          publicKeyHex: Buffer.from(keyPairBls12381G2.publicKey).toString("hex"),
+        })
+        break
+      }
+      default:
+        key = await super.createKey({ type })
+    }
+
+    debug('Created key', type, key.publicKeyHex)
+
+    return key
+  }
+
+  async sign({ keyRef, algorithm, data }: { keyRef: Pick<IKey, 'kid'>; algorithm?: string; data: Uint8Array | Uint8Array[] }): Promise<string> {
+    let privateKey: ManagedPrivateKey
+    try {
+      privateKey = await this.privateKeyStore.get({ alias: keyRef.kid })
+    } catch (e) {
+      throw new Error(`key_not_found: No key entry found for kid=${keyRef.kid}`)
+    }
+
+    if (privateKey.type !== <TKeyType>'Bls12381G2' && !Array.isArray(data)) {
+      return await super.sign({keyRef, algorithm, data})
+    } else if (privateKey.type === <TKeyType>'Bls12381G2') {
+      const keyPair = {
+        keyPair: {
+          secretKey: Uint8Array.from(Buffer.from(privateKey.privateKeyHex, "hex")),
+          publicKey: Uint8Array.from(Buffer.from(keyRef.kid, "hex")),
+        },
+        messages: Array.isArray(data)? data: [data] ,
+      }
+      return Buffer.from(
+        await blsSign(keyPair)
+      ).toString("hex");
+    }
+    throw Error(`not_supported: Cannot sign using key of type ${privateKey.type}`)
+  }
+
+  private asBlsManagedKeyInfo(args: { alias?: string, type: TKeyType, privateKeyHex: string, publicKeyHex: string }): ManagedKeyInfo {
```

Comment:
same comments here, please look at how veramo structures it's plugins and do the same. that way it is alot easier to understand other people's plugins

Reply:
Fixed!